example/lib/main.dart

```diff
@@ -56,269 +68,262 @@ class _MyHomePageState extends State<MyHomePage> {
-        final newToken = json["data"]["token"];
-
-        if (newToken != null) {
-          print("got new token $newToken");
-          preferences.setString("token", newToken);
+        Map json = jsonDecode(event);
+        String? token = json["data"]["token"];
+        if (token != null) {
+          TizenHelperMethods.log("Received a new token: $token");
+          this.token = token;
         }
-      } catch (_) {}
+      } catch (e) {
+        TizenHelperMethods.log("Error parsing message: $e");
+      }
     });
   }
 
   void _pressKey(KeyCodes key) {
-    if (TizenHelperMethods.selectedTv == null) return;
-    print("pressing $key");
-
-    HapticFeedback.lightImpact();
-
+    if (TizenHelperMethods.selectedTv == null) {
+      return;
+    }
+    TizenHelperMethods.log('Sending key: $key');
+    HapticFeedback.mediumImpact();
     TizenHelperMethods.selectedTv!.addToSocket(key);
   }
 
-  bool get connectedToTV =>
-      _connectedTv != null && _connectedTv == TizenHelperMethods.selectedTv;
-
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-        appBar: AppBar(
-          title: DropdownButton<String>(
-            hint: const Text("Select TV"),
-            value: TizenHelperMethods.selectedTv?.name,
-            onChanged: (String? tv) {
-              setState(() {
-                TizenHelperMethods.selectedTv =
-                    tvs.firstWhere((element) => element.name == tv);
-              });
-              setupStream();
-            },
-            items: tvs.map<DropdownMenuItem<String>>((TV tv) {
-              return DropdownMenuItem<String>(
-                value: tv.name,
-                child: Text(
-                  tv.name,
-                  style: const TextStyle(fontSize: 30),
+      appBar: AppBar(
+        title: DropdownButton<String>(
+          hint: const Text("Select TV"),
+          value: TizenHelperMethods.selectedTv?.name,
+          onChanged: (String? tv) {
+            setState(() {
+              TizenHelperMethods.selectedTv =
+                  tvs.firstWhere((element) => element.name == tv);
+            });
+            setupStream();
+          },
+          items: tvs
+              .map<DropdownMenuItem<String>>((TV tv) =>
+                  DropdownMenuItem<String>(
+                    value: tv.name,
+                    child: Text(tv.name, style: const TextStyle(fontSize: 30)),
+                  ))
+              .toList(),
+        ),
+        actions: connectedToTV
+            ? [
+                IconButton(
+                  icon: const Icon(Icons.power_settings_new),
+                  onPressed: () => _pressKey(KeyCodes.keyPower),
                 ),
-              );
-            }).toList(),
-          ),
-          actions: [
-            IconButton(
-              icon: const Icon(Icons.power_settings_new),
-              onPressed: () {
-                _pressKey(KeyCodes.KEY_POWER);
+              ]
+            : [],
```

Comment:
maybe null is representing it better?, it should be the same as the default value.